commons/src/main/java/com/pryv/model/Access.java

```diff
@@ -0,0 +1,332 @@
+package com.pryv.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.pryv.AbstractConnection;
+import com.pryv.database.QueryGenerator;
+import com.pryv.utils.Cuid;
+import com.pryv.utils.JsonConverter;
+import com.rits.cloning.Cloner;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+ * Access data structure from Pryv
+ *
+ * @author tm
+ *
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+public class Access extends ApiResource {
+
+  private String id;
+  private String token;
+  private String name;
+  private ArrayList<Permission> permissions;
+  private Double created;
+  private String createdBy;
+  private Double modified;
+  private String modifiedBy;
+
+  // optional
+  private String type;
+  private String deviceName;
+  private Double lastUsed;
+
+  /**
+   * a weak reference to the connection to which the Access is linked
+   */
+  @JsonIgnore
+  private WeakReference<AbstractConnection> weakConnection;
+
+  /**
+   * used in order to prevent instantiating an Access multiple times.
+   */
+  private static Map<String, Access> supervisor = new WeakHashMap<String, Access>();
+
+  /**
+   * empty Access constructor
+   */
+  public Access() {
+  }
+
+  /**
+   * Constructor for Access object with mandatory fields
+   *
+   * @param name
+   * @param permissions
+   */
+  public Access(String name, ArrayList<Permission> permissions) {
```

Comment:
You should generate an Id for it, no?

Reply:
Id is generated at creation by the API, it does not accept to provide one.

Reply:
Understood, it's true that an Access that hasn't been created on the API doesn't make sense.